### PR TITLE
Update singleCell.go

### PR DESCRIPTION
**Pull request by RavenYL**
Starting to think about ways to reduce memory when using collapseUMI in fastqFilter

# Description

[Description]

## Relevant Links

- [Display Text](https://www.vertgenlab.org/)

### Checklist before requesting a review

- [ ] I performed a self-review of my code
- [ ] If it this a core feature, I have added thorough tests
- [ ] The command `go fmt` was used on all files included

### Testing

None

### Screenshots & Media

None

### Edge cases / Breaking Changes / Known Issues

None

### CODE REVIEW

File: fastq/singleCell.go [hunk 1]
```diff
@@ -63,8 +63,8 @@ func UmiCollision(sc SingleCellPair, umiMap map[string]int) bool {
 	count, found := umiMap[currUmi]
```
Comment (RavenYL) [inline fastq/singleCell.go:63]: Plan:
- run go profiler to see what function takes up lots of memory/time
- maybe a better way to extract barcode (cell barcode, UMI)
- bigger change: because map might be big, might be able to sort fastq (use efficient code in gonomics), and have less look-up (different way to count cell barcodes and UMI)

**Comment by RavenYL** [comment]
go profiler
- go profiler cpu first (as opposed to go profiler memory. More about diagnostics [here on go dev docs](https://go.dev/doc/diagnostics))
- need to add code to cmd file (import, main variable flag, main flag processing, from [this go dev blog post](https://go.dev/blog/pprof))
- (`go test` passed, `go build`) `./fastqFilter -pairedEnd -collapseUmi -cpuprofile=fastqFilter.prof testdata/UmiTest_R1.fastq testdata/UmiTest_R2.fastq out1.fastq out2.fastq` (`rm out1.fastq out2.fastq`, move fastqFilter.prof)
- `go tool pprof fastqFilter.prof` (further interact with .prof using commands like `help` and `top5`)
- Run locally: "Duration: 227.90ms, Total samples = 0" no samples?
- Run on DCC: (`git pull`, `git checkout` branch, `go build`) `cd /work/yl726/raven/shellScripts/fastqFilterTest`, `sbatch --mem=40G --wrap="/hpc/group/vertgenlab/raven/GOPATH/src/github.com/vertgenlab/gonomics/cmd/fastqFilter/fastqFilter -pairedEnd -collapseUmi -cpuprofile=fastqFilter.prof /hpc/group/vertgenlab/raven/raven/IlluminaSequencing/2023-04-05/fastq_allRepsPathsByGem/LibA.R1.V2.fastq.gz /hpc/group/vertgenlab/raven/raven/IlluminaSequencing/2023-04-05/fastq_allRepsPathsByGem/LibA.R2.fastq.gz out1.fastq out2.fastq"`, check `go tool /work/yl726/raven/shellScripts/fastqFilterTest/fastqFilter.prof`

**Comment by RavenYL** [comment]
Note to self: In `cmd/fastqFilter/fastqFilter.go`, `func fastqFilter`, `if s.PairedEnd` then a channel is made, and to collapse UMIs, the `func UmiCollision` in `fastq/singleCell.go` is called, and that function both builds/modifies the umi line by line.

**Comment by RavenYL** [comment]
Closing this fastqFilter_collapseUMI reduce memory GitHub PR since samFilter and cellrangerBam bypass the need to gonomics collapse UMI (filtering is now on the level of sam/bam, and we let cellranger collapse UMI smartly)
